packages/carbon-addons-boomerang-react/src/components/FeatureSideNav/FeatureSideNav.js

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import { settings } from 'carbon-components';
+
+FeatureSideNav.propTypes = {
+  header: PropTypes.func,
+  leftBorder: PropTypes.bool,
+  small: PropTypes.bool,
+  children: PropTypes.any,
+  className: PropTypes.string,
+};
+
+const { prefix } = settings;
+
+export function FeatureSideNav(props) {
+  const { header, leftBorder, small, children, className, listClassName } = props;
+  return (
+    <div className={cx(`${prefix}--bmrg-feature-sidenav-container`, className,  {"--left-border": leftBorder, "--small": small })}>
+      {header && header()}
```

Comment:
Edit: See comment below about adding a list though my comments are still valuable in a different situation

Can you update this so that the header "slot" is not a component function but the invocation of it?
.e.g. pass in `<Header />` instead of `Header` ? This gives us flexibility to pass in whatever props we might need to the Header component.

Reply:
KCD also has a nice blog post that is related to this - warning against calling react components as functions instead of using JSX/React.createElement to render a component https://kentcdodds.com/blog/dont-call-a-react-function-component